File: Dataset/Leetcode/test/101/721.cpp
 class Solution {
public:
    bool XXX(TreeNode* root) {
        if (root == NULL) return true;
        turn(root->right);
        return issame(root->left, root->right);
    }
    void turn(TreeNode* root) {
        if (root == NULL) return;
        if (root->left == NULL && root->right == NULL) return;
        TreeNode *temp = root->left;
        root->left = root->right;
        root->right = temp;
        turn(root->left);
        turn(root->right);
    }
    bool issame(TreeNode *root1, TreeNode *root2) {
        if (root1 == NULL && root2 == NULL)  return true;
        if (root1 == NULL || root2 == NULL)  return false;
        return root1->val == root2->val && issame(root1->left, root2->left) && issame(root1->right, root2->right);
    }
};

File: Dataset/Leetcode/train/88/721.java
 class Solution {
    public void XXX(int[] nums1, int m, int[] nums2, int n) {
        int len = m-- + --n;
        for(int i = len; i >= 0; i--){
            if(m >= 0 && n >= 0){
                if(nums1[m] >= nums2[n])
                    nums1[i] = nums1[m--];
                else
                    nums1[i] = nums2[n--];
            }else if(m < 0)
                nums1[i] = nums2[n--];
        }
    }
}

